process/block/baseProcess.go

```diff
@@ -2342,96 +2303,45 @@ func (bp *baseProcessor) getHeaderHash(header data.HeaderHandler) ([]byte, error
 	return bp.hasher.Compute(string(marshalledHeader)), nil
 }
 
-func (bp *baseProcessor) checkProofRequestingNextHeaderIfMissing(
-	headerShard uint32,
-	headerHash []byte,
-	headerNonce uint64,
-) {
-	if bp.proofsPool.HasProof(headerShard, headerHash) {
-		return
-	}
-
-	log.Debug("could not find proof for header, requesting the next one",
-		"current hash", hex.EncodeToString(headerHash),
-		"header shard", headerShard)
-
-	bp.requestNextHeader(headerHash, headerNonce+1, headerShard)
-}
-
-func (bp *baseProcessor) requestNextHeader(currentHeaderHash []byte, nonce uint64, shardID uint32) {
-	bp.mutRequestedAttestingNoncesMap.Lock()
-	bp.requestedAttestingNoncesMap[string(currentHeaderHash)] = nonce
-	bp.mutRequestedAttestingNoncesMap.Unlock()
-
-	if shardID == core.MetachainShardId {
-		go bp.requestHandler.RequestMetaHeaderByNonce(nonce)
-	} else {
-		go bp.requestHandler.RequestShardHeaderByNonce(shardID, nonce)
+func (bp *baseProcessor) requestProofIfNeeded(currentHeaderHash []byte, epoch uint32, shardID uint32) bool {
+	if !bp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, epoch) {
+		return false
 	}
-}
-
-func (bp *baseProcessor) waitAllMissingProofs(waitTime time.Duration) error {
-	bp.mutRequestedAttestingNoncesMap.RLock()
-	isWaitingForProofs := len(bp.requestedAttestingNoncesMap) > 0
-	bp.mutRequestedAttestingNoncesMap.RUnlock()
-	if !isWaitingForProofs {
-		return nil
+	if bp.proofsPool.HasProof(shardID, currentHeaderHash) {
+		return true
 	}
 
-	select {
-	case <-bp.allProofsReceived:
-		return nil
-	case <-time.After(waitTime):
-		bp.mutRequestedAttestingNoncesMap.RLock()
-		defer bp.mutRequestedAttestingNoncesMap.RUnlock()
+	bp.hdrsForCurrBlock.hdrHashAndInfo[string(currentHeaderHash)].hasProofRequested = true
+	bp.hdrsForCurrBlock.missingProofs++
+	go bp.requestHandler.RequestEquivalentProofByHash(shardID, currentHeaderHash)
 
-		logMessage := ""
-		for hash := range bp.requestedAttestingNoncesMap {
-			logMessage += fmt.Sprintf("\nhash = %s", hex.EncodeToString([]byte(hash)))
-		}
-
-		log.Debug("baseProcessor.waitAllMissingProofs still pending proofs for headers" + logMessage)
-
-		return process.ErrTimeIsOut
-	}
+	return false
 }
 
-func (bp *baseProcessor) checkReceivedHeaderIfAttestingIsNeeded(headerHandler data.HeaderHandler) {
-	if !common.ShouldBlockHavePrevProof(headerHandler, bp.enableEpochsHandler, common.EquivalentMessagesFlag) {
-		return
+func (bp *baseProcessor) checkReceivedProofIfAttestingIsNeeded(proof data.HeaderProofHandler) {
+	bp.hdrsForCurrBlock.mutHdrsForBlock.Lock()
+	hdrHashAndInfo, ok := bp.hdrsForCurrBlock.hdrHashAndInfo[string(proof.GetHeaderHash())]
```

Comment:
here it seems we lost the check for the first transition block to not only have a proof but also have a confirmation block
this was required for the shard as it needed a confirmation (first shard block proof only has 63 consensus size)